dask/dataframe/shuffle.py

```diff
@@ -34,11 +34,19 @@ def set_index(
     if isinstance(index, Series) and index._name == df.index._name:
         return df
     if isinstance(index, (DataFrame, tuple, list)):
-        raise NotImplementedError(
-            "Dask dataframe does not yet support multi-indexes.\n"
-            "You tried to index with this index: %s\n"
-            "Indexes must be single columns only." % str(index)
-        )
+        # Accept ["a"], but not [["a"]]
+        if (
+            isinstance(index, (tuple, list))
```

Comment:
```suggestion
            isinstance(index, list)
```

I'm curious if we should only accept `list` types here.  We could have column names like `("x",)` which we wouldn't want to unpack.

```python
import pandas as pd
df = pd.DataFrame({("x",): [1, 2, 3]})
df.set_index(("x",))
```

This would also allow us to remove the third check in this condition.

It also looks like Pandas doesn't treat a tuple of names as a sequence of names

```python
df = pd.DataFrame({"x": [1, 2, 3]})
df.set_index(["x"])   # Works
df.set_index(("x",))  # Fails
```

Reply:
If you could add the first check as a test that would also be welcome.

Reply:
> We could have column names like ("x",) which we wouldn't want to unpack.

Interesting! I agree. I didn't realize tuples could be used as column names.

That takes be back a step to evaluate (what I think) should be the intended functionality. 
Without this PR:
```python
import pandas as pd
import dask.dataframe as dd

df = pd.DataFrame({
    ('id',)  : ['1', '1', '2', '2', '3'],
    'col1': ['a', 'b', 'a', 'b', 'a']
})
df.set_index(('id',))
dd.from_pandas(df.set_index(('id',)), 1)  # Works
dd.from_pandas(df, 1).set_index(('id',))  # Raises multi-index error
```
I'm assuming that Dask should not be raising the multi-index error when calling `set_index(('id',))`?

Reply:
> I'm assuming that Dask should not be raising the multi-index error when calling set_index(('id',))?

Yes, that seems correct to me.

Reply:
> That takes be back a step to evaluate (what I think) should be the intended functionality.

I do think that we should still respect and support single-element lists, as this PR adds